### PR TITLE
Presentation Exchange: support for json schema filtering and relational constraints

### LINKED ISSUE #350: Presentation Exchange: constraints
Support processing according to the [input evaluation](https://identity.foundation/presentation-exchange/#input-evaluation) section of presentation exchange. Extend current support to allow for the handling of `constraints`.

### PR DESCRIPTION
**Pull request by decentralgabe**
fixes #350, #165, and partially #64

### CODE REVIEW

File: credential/util/util.go [hunk 1]
```diff
@@ -7,59 +7,93 @@ import (
...
 	"github.com/TBD54566975/ssi-sdk/credential/signing"
 	"github.com/goccy/go-json"
+	"github.com/lestrrat-go/jwx/v2/jwt"
 	"github.com/pkg/errors"
 )
 
-// CredentialsFromInterface turn a generic cred into a known shape without maintaining the proof/signature wrapper
-func CredentialsFromInterface(genericCred any) (*credential.VerifiableCredential, error) {
+// ToCredential turn a generic cred into its known object model
+func ToCredential(genericCred any) (*credential.VerifiableCredential, error) {
```
Comment (decentralgabe) [inline credential/util/util.go:15]: @nitro-neal refactored this method and the one below to handle jwt/json properly to fix #165

Comment (nitro-neal) [inline credential/util/util.go:15]: This is awesome, almost need this as a standard for every object.

File: credential/util/util.go [hunk 2]
```diff
@@ -7,59 +7,93 @@ import (
...
 	}
+	return nil, fmt.Errorf("invalid credential type: %s", reflect.TypeOf(genericCred).Kind().String())
 }
 
-// ClaimAsJSON converts a claim with an unknown any into the go-json representation of that credential.
-// claim can only be of type {string, map[string]interface, VerifiableCredential}.
-func ClaimAsJSON(claim any) (map[string]any, error) {
-	switch c := claim.(type) {
+// ToCredentialJSONMap turn a generic cred into a JSON object
+func ToCredentialJSONMap(genericCred any) (map[string]any, error) {
```
Comment (nitro-neal) [inline credential/util/util.go:53]: Alternativly we could make this private and then the master ToCredential funciton will have the signature:

func ToCredential(genericCred any) (*credential.VerifiableCredential, map[string]any, jwt.Token, error) { 

One func to rule them all

**Review by nitro-neal — APPROVED** [review]
Awesome stuff

File: credential/util/util.go (hunk 2 above)
Comment (decentralgabe) [reply credential/util/util.go:53]: I had it like this at first but figured it was cleaner to have each func do a single thing

**Pull request merged**